.github/workflows/kind_host.yml

```diff
@@ -0,0 +1,130 @@
+name: Framework KinD Hosted mode Workflow
+
+on:
+  workflow_call:
+    inputs: # To specify where they're located for components, it's leveraged by our controller repos.
+      hub_only_component:
+        required: false
+        type: string
+        default: 'false'
+
+env:
+  RELEASE_BRANCH: ${{ github.event.pull_request.base.ref || github.ref_name }}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  tests:
+    name: Hosted mode Tests # Part of the check name, be careful when changing.
+
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: localhost:5000
+    strategy:
+      # Don't skip all tests if one fails - others may still have useful info
+      fail-fast: false
+
+    steps:
+    - name: Checkout Component Repository
+      # Checkout a specific component when called from another repository
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        # `repository` is inferred as the "caller" repository 
+        path: component
+        # `ref` is inferred as the new commit in the "caller" repository
+
+    - name: Checkout Policy Framework
+      # Checkout a stable branch when called from another repository
+      if: ${{ github.event.repository.name != 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        repository: stolostron/governance-policy-framework
+        path: framework
+        ref: ${{ env.RELEASE_BRANCH }} # like main or release-*
+
+    - name: Checkout Policy Framework
+      # Checkout like "usual" when called from this repository
+      if: ${{ github.event.repository.name == 'governance-policy-framework' }}
+      uses: actions/checkout@v3
+      with:
+        # `repository` is inferred as the "caller" repository 
+        path: framework
+        # `ref` is inferred as the new commit
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version-file: framework/go.mod
+
+    - name: Bootstrap the KinD Cluster for hosted mode
+      working-directory: framework
+      run: |
+        echo "::group::make e2e-dependencies"
+        make e2e-dependencies
+        echo "::endgroup::"
+
+        echo "::group::make setup hosted mode"
+        make kind-bootstrap-hosted
+        echo "::endgroup::"
+        echo "::group::Saving kubeconfig paths for use in other steps"
+    
+        echo "MANAGED_KUBECONFIG=$(pwd)/kubeconfig_managed" >> $GITHUB_ENV
+        echo "HUB_KUBECONFIG=$(pwd)/kubeconfig_hub" >> $GITHUB_ENV
+        echo "HUB_INTERNAL_KUBECONFIG=$(pwd)/kubeconfig_hub_internal" >> $GITHUB_ENV
+        echo "::endgroup::"
+    
+        echo "::group::test for patch step"
+        export MANAGED_CONFIG=$(pwd)/kubeconfig_managed
+        export HUB_CONFIG=$(pwd)/kubeconfig_hub
+        export HUB_CONFIG_INTERNAL=$(pwd)/kubeconfig_hub_internal
+        echo "::endgroup::"
+
+    - name: Patch Component Image
```

Comment:
I don't think this will work in hosted mode. I think you likely want to run `make build-images` and then run something like `kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)` on the Hub cluster so that the new image is used. You may have to manually trigger a "rollout" of the controller here as well.